week1/homework/src/server.js

```diff
@@ -5,16 +5,42 @@ const http = require('http');
 /* `createServer` MUST return an instance of `http.Server` otherwise the tests
  * will fail.
  */
-function createServer(port) {
+function createServer() {
   let state = 10;
-
   const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
+    switch (request.url) {
+      case '/state':
+        response.setHeader('Content-Type', 'application/json');
+        const stringifiedState = JSON.stringify({ state: state });
+        response.write(stringifiedState);
+        break;
+      case '/add':
+        response.setHeader('Content-Type', 'application/json');
+        const stringifiedState2 = JSON.stringify({ state: ++state });
+        response.write(stringifiedState2);
+        break;
+      case '/subtract':
+        response.setHeader('Content-Type', 'application/json');
+        const stringifiedState3 = JSON.stringify({ state: --state });
```

Comment:
PRO TIP: Does this variable serve a real purpose? If you find yourself in a situation where you need to call your variable xyz3 simply because 1 and 2 are already used perhaps it is time to do some refactoring